subcommands/diff/diff.go

```diff
@@ -43,59 +47,87 @@ func (cmd *Diff) Parse(ctx *appcontext.AppContext, args []string) error {
 		fmt.Fprintf(flags.Output(), "\nOPTIONS:\n")
 		flags.PrintDefaults()
 	}
-
 	flags.BoolVar(&cmd.Highlight, "highlight", false, "highlight output")
+	flags.BoolVar(&cmd.Recursive, "recursive", false, "recursive diff of directories")
 	flags.Parse(args)
 
-	if flags.NArg() != 2 {
-		return fmt.Errorf("needs two snapshot ID and/or snapshot files to diff")
+	if flags.NArg() == 1 {
+		cmd.Path1 = flags.Arg(0)
+		cmd.Path2 = ""
+	} else if flags.NArg() == 2 {
+		cmd.Path1 = flags.Arg(0)
+		cmd.Path2 = flags.Arg(1)
+	} else {
+		return fmt.Errorf("needs at least a snapshot ID and/or snapshot file to diff")
 	}
-
 	cmd.RepositorySecret = ctx.GetSecret()
-	cmd.SnapshotPath1 = flags.Arg(0)
-	cmd.SnapshotPath2 = flags.Arg(1)
 
 	return nil
 }
 
 type Diff struct {
 	subcommands.SubcommandBase
 
-	Highlight     bool
-	SnapshotPath1 string
-	SnapshotPath2 string
+	RepositoryLocation string
+	RepositorySecret   []byte
```

Comment:
i don't think we need this, is it a merge thing from an old branch?  aren't they in subcommand base nowadays?